### DISCUSSION TITLE
Is it possible to use public CDN cache for not-authenticated users and use private cache for authenticated users for the same route

### DISCUSSION THREAD

**Discussion by MartinXPN** [vercel/next.js#39794] [Help]
## The question
Is it possible to store a public cache by setting an HTTP header `Cache-Control` as described in docs: https://nextjs.org/docs/going-to-production#caching for users that are not authenticated, while having a different cache/page for users that are authenticated?

## My use case
* I use firebase authentication: https://github.com/gladly-team/next-firebase-auth
* When accessing the homepage, I'd like to:
    * Show the landing page for users that are not authenticated
    * Show the dashboard to users that have been authenticated.
* The landing page basically doesn't change - so it would be very helpful to cache it to reduce redundant requests to the server.
* The dashboard is specific to each user - so it can be both rendered on the server-side and on the client-side.
* It would be perfect to have the same route for those two (`/`) and not redirect the authenticated user to a different route.

------------

Any advice, workaround, tip, or suggestion would be really appreciated!

**Comment by gaston-flores** [comment]
There's no straightforward way of doing this. You can tell a cache to not store private routes, but the cache has no way of knowing if the user is authenticated so it doesn't know whether to serve the static front page or not.

[Combining static generation with getServerSideProps](https://github.com/vercel/next.js/discussions/11424) would be useful for your use case, but it's something the Next team is still working on. I think the least intrusive option would be:

- Use `getStaticProps` for your front page
- Put the dashboard under `/dashboard`
- Use a [middleware ](https://nextjs.org/docs/advanced-features/middleware)to check if the user is authenticated and redirect them to `/dashboard` in that case.

**Reply by MartinXPN** [reply]
Thanks for the answer @gaston-flores !
AWS (CloudFront) allows you to configure [CDN caching by a specific cookie](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/Cookies.html) (in this case it can be an auth-id). Yet, seems like Vercel doesn't provide that option.

I really don't want to redirect the user to another route like `/dashboard`, so doing this with middleware wouldn't work for me.

I guess the only option remains to just keep the `Cache-Control` header `private`.
